### PR TITLE
TD-875: Add Custom Language form and handler

### PR DESCRIPTION
**Pull request by branson-perreault**
@vendasta/technically-developers 

### Problem

We want to allow partners to come into Weblate and perform translations/text changes without needing to manually grant them permissions. That means we need to isolate any changes they make.

### Solution

- Allow partners to create "custom" translations based off existing translations. 
  - This is accomplished by generating new languages on the fly using a patterned language code: `en_US~VUNI`
  - New languages are added to the partner "namespace" permissions group.
- Changes from the base translations will continue to propagate to the custom translations, unless a change has already been made by the partner.

### CODE REVIEW

File: weblate/vendasta/access.py [hunk 1]
```diff
@@ -35,4 +35,10 @@ def get_or_create_namespace_group(namespace):
     namespace_group = Group.objects.get_or_create(name=namespace.upper())
     if isinstance(namespace_group, tuple):
         namespace_group = namespace_group[0]
+
+    access_namespace_role = Role.objects.get_or_create(name="Access Namespace")
```
Comment (bbudz-va) [inline weblate/vendasta/access.py:39]: could `Access Namespace` be a constant?

Comment (bperreault-va) [reply weblate/vendasta/access.py:39]: Definitely a good idea.

File: weblate/lang/models.py [hunk 2]
```diff
@@ -108,6 +109,10 @@ def try_get(self, *args, **kwargs):
 
     def parse_lang_country(self, code):
         """Parse language and country from locale code."""
+        # Strip namespace
+        if NAMESPACE_SEPARATOR in code:
+            code = code.split(":")[0]
```
Comment (rrance-va) [inline weblate/lang/models.py:114]:
```suggestion
            code = code.split(NAMESPACE_SEPARATOR)[0]
```

File: weblate/trans/views/basic.py [hunk 3]
```diff
@@ -351,11 +361,61 @@ def new_language(request, project, component):
...
+@login_required
+def new_namespaced_language(request, project, component):
+    obj = get_component(request, project, component)
+
+    namespace = ""
+    namespace_query = request.user.groups.filter(roles__name=ACCESS_NAMESPACE).order_by(
+        "name"
     )
+    if namespace_query.count():
+        namespace = namespace_query[0]
```
Comment (rrance-va) [inline weblate/trans/views/basic.py:385]: Is it ever possible for a user to be in multiple partner groups? Do super admins need access to create namespaced translations on behalf of a partner?

Comment (bperreault-va) [reply weblate/trans/views/basic.py:385]: I believe it's technically possible, but only for internal users. We may need to create namespaced translations on behalf of a partner, and perhaps we can use query parameters for that when the time comes.

File: weblate/vendasta/auth.py [hunk 4]
```diff
@@ -15,14 +15,15 @@ class VendastaOpenIdConnect(OpenIdConnectAuth):
 
     @cache(ttl=86400)
     def oidc_config(self):
-        oidc_endpoint = self.setting("OIDC_ENDPOINT", "http://iam-prod.vendasta-internal.com")
+        oidc_endpoint = self.setting(
+            "OIDC_ENDPOINT", "http://iam-prod.vendasta-internal.com"
```
Comment (bbudz-va) [inline weblate/vendasta/auth.py:19]: this is ok to always be pointed at prod?

File: k8s/deployment-demo.yml [hunk 5]
```diff
@@ -0,0 +1,282 @@
+apiVersion: extensions/v1beta1
```
Comment (bbudz-va) [inline k8s/deployment-demo.yml:1]: feeling out of my depth with this file 😅  there doesn't need to be a config for prod though?

Comment (bperreault-va) [reply k8s/deployment-demo.yml:1]: This is a duplicate of the production deployment.yaml file, with values adjusted for demo!

File: weblate/vendasta/auth.py (hunk 4 above)
Comment (bperreault-va) [reply weblate/vendasta/auth.py:19]: The second parameter is a fallback for if the "OIDC_ENDPOINT" configuration is not found.

Comment (bbudz-va) [reply weblate/vendasta/auth.py:19]: ah, gotcha! 👍

**Pull request merged**